llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -6053,9 +6053,38 @@ Value *SPIRVToLLVM::transSPIRVImageQueryLodFromInst(SPIRVInstruction *bi, BasicB
   // A sampler descriptor is encoded as {desc,convertingSamplerIdx}. Extract the actual sampler.
   Value *samplerDesc = getBuilder()->CreateExtractValue(imageInfo.samplerDesc, 0);
 
-  // Generate the operation.
+  // Generate the operation for normal image get lod.
   Value *coord = transValue(bii->getOpValue(1), bb->getParent(), bb);
-  return getBuilder()->CreateImageGetLod(imageInfo.dim, imageInfo.flags, imageInfo.imageDesc, samplerDesc, coord);
+  Value *result =
+      getBuilder()->CreateImageGetLod(imageInfo.dim, imageInfo.flags, imageInfo.imageDesc, samplerDesc, coord);
+
+  // Extract sampler descriptor for ycbcr, otherwise the desc would be null.
+  if (!m_convertingSamplers.empty()) {
+    // Refer to transSPIRVImageSampleFromInst for details.
```

Comment:
can you flip this condition to return early is `m_convertingSamplers` is empty?

Reply:
I assume no. 
For that the logic is, if m_convertingSamplers  is empty, we need the result for normal image get lod. So no matter m_convertingSamplers  is empty or not, we need to call getBuilder()->CreateImageGetLod once.

Reply:
I guess Jakub is suggesting something like

```
if (m_convertingSamplers.empty())
  return result;
```

then the rest of the code does not need to be indented. I think that will work, right?

Reply:
Okay, but with the updated version, i think it's fine now.

Reply:
Looks better to me now, thanks